File: gnm-run/mainSlurmCombine.cpp
//
// Created by Max Würfel on 27.06.23.
//

#include <string>
#include "Slurm.h"
#include <vector>

int main() {
    bool cluster = false;
    std::string inDirPath;

    if (cluster) {
        inDirPath = "/store/DAMTPEGLEN/mw894/slurm";
    } else {
        inDirPath = "/Users/maxwuerfek/code/diss/gnm-run/slurm";
    }

    std::vector<std::vector<std::vector<std::vector<double>>>> Kall;
    std::vector<std::vector<double>> paramSpace;
    std::vector<std::string> groupIds;
    Slurm::combineResFiles(inDirPath, Kall, paramSpace, groupIds);
    Slurm::writeGroupsHDF5(groupIds, Kall, paramSpace, inDirPath);
    return 0;
}

File: gnm-run/mainSlurmCombine.py
import numpy as np
from Slurm import Slurm

dSets = ["Charlesworth2015", "Hennig2011", "Demas2006"]


def main():
    cluster = False
    dSet = 0

    if cluster:
        inDirPath = "/store/DAMTPEGLEN/mw894/slurm/" + dSets[dSet]
    else:
        inDirPath = "/Users/maxwuerfek/code/diss/slurm/" + dSets[dSet]

    Kall, paramSpace, groupIds, filesRead = Slurm.combineResFiles(inDirPath)
    print("Files read: " + str(filesRead))
    Slurm.writeGroupsHD5(Kall, paramSpace, groupIds, inDirPath)
    print("Groups written:" + str(len(np.unique(groupIds))))


if __name__ == "__main__":
    main()
